### ISSUE TITLE
Available balance is not checked against outgoing pending STX transfers and fees

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#5497]
Steps to reproduce:
1. Click send in the wallet
2. Select Stacks
3. Click send max
4. transmit
5. Click send in the wallet
6. Select Stacks (even though there is 0 balance)
7. Click send max
8. transmit
9. See error, the transaction is transmitted without any error getting thrown. 
A negative balance is shown here, but not in the lower right corner on the "send" flow. 
![image](https://github.com/leather-wallet/extension/assets/33360391/67ba5c0b-8799-481a-89b3-02554c6017e2)

https://github.com/leather-wallet/extension/assets/33360391/b6de301f-0065-4e50-a2c8-8a682b31389e

![image](https://github.com/leather-wallet/extension/assets/33360391/64ae527d-7131-4158-b48b-00ac72dc6740)

An additional issue, related, when increasing the fee available balance after adjusting for pending tx (outgoing STX and fees) is also not checked. I can click transmit 
![image](https://github.com/leather-wallet/extension/assets/33360391/969cc3a9-8941-4758-9f87-e3e486bd1e1a)

even though the address doesn't have the balance to cover this. After clicking transmit, it turns out it isn't actually transmitted: I see an error thrown "Not enough funds" and:
![image](https://github.com/leather-wallet/extension/assets/33360391/2808e034-c997-47e8-8491-68c2acddd953)

But I expect the user to be shown a warning rather than letting them transmit the impossible transaction. 

Expected
I expect the user to be prevented from creating a new transaction and shown guidance on why it is prevented. Not enough funds to cover transfer + fee.

**Comment by markmhendrickson** [comment]
@alter-eggo any idea if this is a regression caused by other recent balance-related work?

**Comment by alter-eggo** [comment]
@markmhendrickson yeah, might be caused by recent refactoring of stacks related code, need to investigate

**Comment by 314159265359879** [comment]
The check is still failing, but this issue is less of a priority now that transactions are processed in seconds rather then minutes. A negative balance is no longer displayed. 

Balance shows 0 STX now, not a negative value which is an improvement. 
![Image](https://github.com/user-attachments/assets/f2382eb9-bbdf-4616-b0cf-bda481701359)

**Comment by markmhendrickson** [comment]
@alter-eggo is it worth tackling this now given your work fixing balances generally with [LEA-1941](https://linear.app/leather-io/issue/LEA-1941/resolve-bugs-with-total-and-available-balances-for-various-cases-and) ? cc @camerow

**Comment by camerow** [comment]
@314159265359879 not great still, I would say it's not even an improvement just a less emotionally charged bad value.

@alter-eggo this is pretty core behavior, this needs a fix ASAP
